tests/__init__.py

```diff
@@ -1,24 +0,0 @@
-import os
-import sys
-import unittest
-
-if os.name == "nt" and sys.version_info[1] >= 8:
```

Comment:
Were lines 5-8 and 22-24 not copied to to `shapely/tests/legacy/__init__.py` because they are no longer used?  Apologies if I missed the comment describing these changes.

Reply:
> Were lines 5-8 and 22-24 not copied to to `shapely/tests/legacy/__init__.py` because they are no longer used? Apologies if I missed the comment describing these changes.

I recall so but I can't find the reference either. Should I restore them @jorisvandenbossche?

Reply:
From what I can tell, `shapely20_deprecated` is not used anywhere else, so I'm OK with dropping that here.

I don't have the ability to judge the impact of setting the GEOS DLL directory for Windows, but I'm assuming since all the Windows tests still pass that it isn't needed under a standard installation, and the operation of loading it isn't used anywhere else in the codebase.  So - I have no objections to removing this either.

Just wanted to be sure I was following the reason they we modified; all other changes here were expected and easy to review.  Thanks for the multiple rounds of refactoring & rebasing @idanmiara !

Reply:
Yes, I don't think that section is needed either as there are many tests that passed on windows also before this change and all the tests pass also after this change...
I also tested it on windows myself